src/main/java/wtf/g4s8/examples/Main.java

```diff
@@ -77,7 +77,7 @@ public static void main(final String... args) throws Exception {
                 }
                 try {
                     final String res = proposers.get(val).propose(Integer.toString(val)).get();
-                    System.out.printf("proposed %d get %s\n", val, res);
+                    System.out.printf("[" + Thread.currentThread().getName() + "]" + "proposed %d get %s\n", val, res);
```

Comment:
@stepanvalyavskiy maybe we can create some simple logging tool for printing such messages?